workflow-templates/im-deploy-on-prem-website.yml

```
@@ -0,0 +1,462 @@
+      # fetch-depth: 0 to retrieve the history and tags so we can check if a tag is reachable from the default branch.
+      - name: Checkout Repository
+        if: needs.set-vars.outputs.ENVIRONMENT == 'prod'
+        uses: actions/checkout@v2
+        with:
+          ref: 'main' # TODO: verify the name of your default branch
+          fetch-depth: 0
+
+      - uses: im-open/is-tag-reachable-from-default-branch@v1.0.0
+        if: needs.set-vars.outputs.ENVIRONMENT == 'prod'
+        with:
+          tag: ${{ env.GITHUB_REF }}
+
+  build-deploy:
+    needs: [set-vars, stakeholder-approval, validate-tag-for-prod-deploys]
+    runs-on: [self-hosted, windows-2019]
+    environment: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+
+    env:
+      PAGERDUTY_SERVICE_IDS: '[ "${{ secrets.PAGERDUTY_SERVICE_ID }}" ]' # TODO:  Add your PD Service IDs here, the svc id usually has a format of P0XXXXX
+      PAGERDUTY_WINDOW_IN_MIN: 30 # TODO:  Verify the length of your PD Maintenance Window
+      PAGERDUTY_WINDOW_DESC: 'Deploying Code to ${{ needs.set-vars.outputs.ENVIRONMENT }} from GitHub Actions' # TODO:  Verify this PD Maintenance Window Description
+      DOTNET_VERSION: '' # TODO:  Set the .net core version of your project
+      PROJECT_ROOT: '' # TODO:  set this to the folder of your web site project. For most projects this will be ./src/ProjectName
+      DEPLOY_ZIP: 'published_site.zip'
+
+    steps:
+      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}"
+
+      - name: Open a PagerDuty Maintenance Window
+        id: open-window
+        uses: im-open/open-pagerduty-maintenance-window@v1.0.0
+        with:
+          pagerduty-api-key: ${{ secrets.PAGERDUTY_API_KEY }}
+          description: '${{ env.PAGERDUTY_WINDOW_DESC }}'
+          minutes: ${{ env.PAGERDUTY_WINDOW_IN_MIN }}
+          service-ids: ${{ env.PAGERDUTY_SERVICE_IDS }}
+
+      - run: |
+          echo "The maintenance window ID is: ${{ steps.open-window.outputs.maintenance-window-id }}"
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ env.GITHUB_REF }}
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+
+      # TODO:  If you need to do replacements in your appsettings.json file you can do it two ways.
+      # 1- Repeat this add-or-update-json-properties action for each environment, and add an 'if: needs.set-vars.outputs.ENVIRONMENT == ""' condition with the appropriate env to each action.
+      # 2- In the set-vars job, add one set-variable-based-on-environment action block per variable then use that variable in the single add-or-update-json-properties action below.
+      # - name: Update appsettings.json with Environment Specific Values
+      #   uses: im-open/add-or-update-json-properties@v1.0.1
+      #   with:
+      #     path-to-json-file: '${{ env.PROJECT_ROOT }}/appsettings.json'
```

Pre-built artifact change: 
```suggestion
      #     path-to-json-file: '${{ env.UNZIPPED_ASSET }}/appsettings.json'
```